server/bleep/src/semantic/embedder.rs

```diff
@@ -0,0 +1,271 @@
+        let token_type_ids = tokenizer_output.get_type_ids();
+        let length = input_ids.len();
+        trace!("embedding {} tokens {:?}", length, sequence);
+
+        let inputs_ids_array = ndarray::Array::from_shape_vec(
+            (1, length),
+            input_ids.iter().map(|&x| x as i64).collect(),
+        )?;
+
+        let attention_mask_array = ndarray::Array::from_shape_vec(
+            (1, length),
+            attention_mask.iter().map(|&x| x as i64).collect(),
+        )?;
+
+        let token_type_ids_array = ndarray::Array::from_shape_vec(
+            (1, length),
+            token_type_ids.iter().map(|&x| x as i64).collect(),
+        )?;
+
+        let outputs = self.session.run([
+            InputTensor::from_array(inputs_ids_array.into_dyn()),
+            InputTensor::from_array(attention_mask_array.into_dyn()),
+            InputTensor::from_array(token_type_ids_array.into_dyn()),
+        ])?;
+
+        let output_tensor: OrtOwnedTensor<f32, _> = outputs[0].try_extract().unwrap();
+        let sequence_embedding = &*output_tensor.view();
+        let pooled = sequence_embedding.mean_axis(Axis(1)).unwrap();
+        Ok(pooled.to_owned().as_slice().unwrap().to_vec())
+    }
+
+    fn tokenizer(&self) -> &Tokenizer {
+        &self.tokenizer
+    }
+
+    async fn batch_embed(&self, log: &EmbedLog, _flush: bool) -> anyhow::Result<Vec<PointStruct>> {
+        let mut output = vec![];
+        while let Some(entry) = log.pop() {
+            output.push(PointStruct {
+                id: Some(PointId::from(entry.id)),
+                vectors: Some(
+                    tokio::task::block_in_place(|| self.embed(entry.data.as_ref()))?.into(),
+                ),
+                payload: entry.payload,
+            });
+        }
+
+        Ok(output)
+    }
+}
+
+pub struct RemoteEmbedder {
+    url: reqwest::Url,
+    session: reqwest::Client,
+    embedder: LocalEmbedder,
+}
+
+impl RemoteEmbedder {
+    pub fn new(url: reqwest::Url, model_dir: &Path) -> anyhow::Result<Self> {
+        let url = url.join("encode")?;
+        Ok(Self {
+            url,
+            session: reqwest::Client::new(),
+            embedder: LocalEmbedder::new(model_dir)?,
+        })
+    }
+
+    async fn make_request(&self, request: ServerRequest<'_>) -> anyhow::Result<ServerResponse> {
+        Ok(self
+            .session
+            .post(self.url.clone())
+            .json(&request)
+            .send()
+            .await?
+            .json()
+            .await?)
+    }
+}
+
+#[async_trait]
+impl Embedder for RemoteEmbedder {
+    fn embed(&self, data: &str) -> anyhow::Result<Embedding> {
+        self.embedder.embed(data)
+    }
+
+    fn tokenizer(&self) -> &Tokenizer {
+        self.embedder.tokenizer()
+    }
+
+    async fn batch_embed(&self, log: &EmbedLog, flush: bool) -> anyhow::Result<Vec<PointStruct>> {
+        const MAX_BATCH_SIZE: usize = 128;
```

Comment:
This will change depending on server configuration. This should be a command line argument that is configured in Helm. Can use this as a default value.